files.qmd

```diff
@@ -2,37 +2,74 @@
 
 ## Names
 
-File names should be meaningful and end in `.R`. Avoid using special characters
-in file names - stick with numbers, letters, `-`, and `_`.
-
+1.  File names should be **machine readable**: avoid spaces, symbols, and special characters. Prefer file names that are all lower case, and never
+have names that differ only in their capitalization. Use `.R` as the extension of R files.
 
+    ```
     # Good
     fit_models.R
     utility_functions.R
+    exploratory-data-analysis.R
 
     # Bad
     fit models.R
     foo.r
-    stuff.r
+    ExploratoryDataAnaylsis.r
+    ```
+
+2.  File names should be **human readable**: use file names to describe what's in the file.
+
+    ```
+    # good
+    report-draft-notes.txt
+
+    # bad
+    temp.r
+    ```
+
+    Use the same structure for closely related files:
 
-If files should be run in a particular order, prefix them with numbers. If it
-seems likely you'll have more than 10 files, left pad with zero:
+    ```
+    # good
+    figure-01.png
+    figure-02.png
```

Comment:
> start file names with numbers so that alphabetical sorting puts them in the order they get used

quoted from below, goes against this good example